muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -170,45 +610,180 @@ public Response listNotesByOwner(@PathParam("OWNER") Long owner, @QueryParam("li
+      NoteRestModel noteRest = toRestModel(note);
+
+      for (NoteReceiver recipient : recipients) {
+        NoteReceiverRestModel recipientRest = toRestModel(recipient);
+
+        recipientListRest.add(recipientRest);
+      }
+      noteRest.setRecipients(recipientListRest);
+      noteRest.setMultiUserNote(recipientListRest.size() > 1);
+
+      notesList.add(noteRest);
+    }
+
     return Response.ok(notesList).build();
   }
-  
+
   // mApi() call (mApi().notes.note.toggleArchived.update(noteId))
-  // In this case archiving means moving note to 'trash'. So it's only update method and user can restore the note.
+  // In this case archiving means moving note to 'trash'. So it's only update
+  // method and user can restore the note.
   @PUT
-  @Path ("/note/{NOTEID}/toggleArchived")
-  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
-  public Response toggleArchived(@PathParam ("NOTEID") Long noteId) {
+  @Path("/note/{NOTEID}/toggleArchived")
+  @RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)
+  public Response toggleArchived(@PathParam("NOTEID") Long noteId) {
     Note note = notesController.findNoteById(noteId);
-    
+
     if (note == null) {
       return Response.status(Status.NOT_FOUND).entity(String.format("Note (%d) not found", noteId)).build();
     }
-    
+
     // Students can only archive their own notes
-    if (sessionController.hasRole(EnvironmentRoleArchetype.STUDENT) && !note.getCreator().equals(sessionController.getLoggedUserEntity().getId())) {
-      return Response.status(Status.FORBIDDEN).build(); 
+    if (sessionController.hasRole(EnvironmentRoleArchetype.STUDENT)
+        && !note.getCreator().equals(sessionController.getLoggedUserEntity().getId())) {
+      return Response.status(Status.FORBIDDEN).build();
     }
-    
+
+    List<NoteReceiver> recipients = noteReceiverController.listByNote(note);
+    NoteReceiver receiver = null;
+
     // Archiving is only allowed if you're the creator or owner of the note
     if ((!sessionController.getLoggedUserEntity().getId().equals(note.getCreator()))) {
-      if (!sessionController.getLoggedUserEntity().getId().equals(note.getOwner())) {
+
+      if (recipients.size() == 1) {
+        receiver = recipients.get(0);
+      }
+      if (receiver == null || !sessionController.getLoggedUserEntity().getId().equals(receiver.getId())) {
         return Response.status(Status.FORBIDDEN).build();
       }
     }
 
     NoteRestModel noteRestModel = toRestModel(notesController.toggleArchived(note));
+
+    List<NoteReceiverRestModel> recipientsRest = new ArrayList<NoteReceiverRestModel>();
+    for (NoteReceiver recipient : recipients) {
+      NoteReceiverRestModel recipientRest = toRestModel(recipient);
+
+      recipientsRest.add(recipientRest);
+    }
+
+    noteRestModel.setRecipients(recipientsRest);
+    noteRestModel.setMultiUserNote(recipientsRest.size() > 1);
```

Comment:
Once again, what function does `note.getMultiUserNote()` serve if a multi-user note depends on the number of its recipients? Multiple occurences in this class.